### DISCUSSION TITLE
build iOS failed

### DISCUSSION THREAD

**Discussion by Anyewuya** [alibaba/xquic#21] [Q&A] [answered]
macOS 11.4
Xcode  12.5
cmake version 3.21.2

1.export IOS_CMAKE_TOOLCHAIN="xx/xquic/cmake/ios.toolchain.cmake"
2. sh xqc_build.sh ios ios_build ios_release xxx/xquic/third_party/boringssl

output:
[  1%] Generating crypto_test_data.cc
make[2]: *** No rule to make target `xxx/xquic/bssl_symbols.txt', needed by `third_party/boringssl/symbol_prefix_include/boringssl_prefix_symbols.h'.  Stop.
make[1]: *** [third_party/boringssl/CMakeFiles/boringssl_prefix_symbols.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  1%] Building CXX object third_party/boringssl/CMakeFiles/boringssl_gtest.dir/third_party/googletest/src/gtest-all.cc.o
[  1%] Building CXX object third_party/boringssl/CMakeFiles/crypto_test_data.dir/crypto_test_data.cc.o
[  1%] Built target crypto_test_data
[  1%] Linking CXX static library ../../outputs/libboringssl_gtest.a
[  1%] Built target boringssl_gtest
make: *** [all] Error 2

**Comment by ruiqizhou** [comment]
did Boringssl compile successfully?

**Reply by Anyewuya** [reply]
compile success.
➜  build git:(master) cmake -DBUILD_SHARED_LIBS=0 -DCMAKE_C_FLAGS="-fPIC" -DCMAKE_CXX_FLAGS="-fPIC" ..
make ssl crypto
-- Configuring done
-- Generating done
-- Build files have been written to:xxx/xquic/third_party/boringssl/build
[  0%] Built target global_target
Scanning dependencies of target fipsmodule
Consolidate compiler generated dependencies of target fipsmodule
[ 10%] Built target fipsmodule
Scanning dependencies of target crypto
Consolidate compiler generated dependencies of target crypto
[ 87%] Built target crypto
Consolidate compiler generated dependencies of target ssl
[100%] Built target ssl
[  0%] Built target global_target
[ 11%] Built target fipsmodule
[100%] Built target crypto

**Comment by ruiqizhou** [comment, marked as answer]
Sorry, it seems we have some problems with `xquic_build.sh` and we will fix them as soon as possible.

However, if you want a temporary solution to this problem, you can remove the contents of the red box. Note that this approach takes away the ability of boringssl to add symbolic prefixes.
![image](https://user-images.githubusercontent.com/95687026/148936541-05147f0c-a5cc-41fa-9d06-2107a57c7772.png)

**Reply by Anyewuya** [reply]
👍 build ok.

**Reply by ruiqizhou** [reply]
ok, we have opened an issue for this problem #33.

**Comment by Anyewuya** [comment]
编译好后，构建目录中有多个静态库，是正常的吗。
要在 demo 中测试，需要把红框中所有的静态库都加上？
![image](https://user-images.githubusercontent.com/6964964/149051604-ff8068ab-89c6-4a2a-b1ef-9ed975f74d53.png)

**Reply by ruiqizhou** [reply]
`/ios/xquic/xquic`

**Reply by lurker-Chen** [reply]
会在/ios/xquic/xquic目录下生成整合的lib和headers，在项目里依赖这个

**Comment by LevenAn** [comment]
New problem 😄 ！when i build with “sh xqc_build.sh ios ios_build ios_release ./third_party/boringssl”
Error：
CMake Error at third_party/boringssl/tool/CMakeLists.txt:24 (install):
  install TARGETS given no BUNDLE DESTINATION for MACOSX_BUNDLE executable
  target "bssl".

**Reply by yifzone** [reply]
some problem

**Reply by kelvin943** [reply]
some problem +1
